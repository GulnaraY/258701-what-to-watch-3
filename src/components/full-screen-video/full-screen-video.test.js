import React from 'react';
import rerender from 'react-test-renderer';
import FullScreenVideo from './full-screen-video.jsx';

const PromoMovieDetails = {
  TITLE: `Friends`,
  VIDEO: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  POSTER: `moonrise-kingdom.jpg`,
};


it(`Should render FullScreenVideo`, () => {
  const tree = rerender.create(
      <FullScreenVideo
        onVideoExit={()=>{}}
        title={PromoMovieDetails.TITLE}
        video={PromoMovieDetails.VIDEO}
        poster={PromoMovieDetails.POSTER}
      />, {
        createNodeMock: () => {
          return {};
        }
      }
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
