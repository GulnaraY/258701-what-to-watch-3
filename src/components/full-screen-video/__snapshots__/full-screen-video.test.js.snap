// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render FullScreenVideo 1`] = `
<div
  className="player"
>
  <video
    height="100%"
    muted={false}
    poster="moonrise-kingdom.jpg"
    width="100%"
  >
    <source
      src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
      type="video/mp4"
    />
  </video>
  <button
    className="player__exit"
    onClick={[Function]}
    type="button"
  >
    Exit
  </button>
  <div
    className="player__controls"
  >
    <div
      className="player__controls-row"
    >
      <div
        className="player__time"
      >
        <progress
          className="player__progress"
          max="100"
          value={0}
        />
        <div
          className="player__toggler"
          style={
            Object {
              "left": "0%",
            }
          }
        >
          Toggler
        </div>
      </div>
      <div
        className="player__time-value"
      >
        0.1
      </div>
    </div>
    <div
      className="player__controls-row"
    >
      <button
        className="player__play"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="21"
          viewBox="0 0 14 21"
          width="14"
        >
          <use
            xlinkHref="#play"
          />
        </svg>
        <span>
          Pause
        </span>
      </button>
      <div
        className="player__name"
      >
        Friends
      </div>
      <button
        className="player__full-screen"
        type="button"
      >
        <svg
          height="27"
          viewBox="0 0 27 27"
          width="27"
        >
          <use
            xlinkHref="#full-screen"
          />
        </svg>
        <span>
          Full screen
        </span>
      </button>
    </div>
  </div>
</div>
`;
